src/ReplaceTransition.js

```diff
@@ -28,7 +28,13 @@ class ReplaceTransition extends React.Component {
     const child = React.Children.toArray(children)[idx];
 
     if (child.props[handler]) child.props[handler](...originalArgs)
-    if (this.props[handler]) this.props[handler](ReactDOM.findDOMNode(this))
+    if (this.props[handler]) {
+      const nodeRef = idx === 0
+        ? this.props.firstNodeRef
+        : this.props.secondNodeRef
```

Comment:
This depends on how `nodeRef` is used on children transition.

If user will use this way, then we can extract the ref from children

```jsx
function App() {
  const firstNodeRef = useRef()
  const secondNodeRef = useRef()
  return (
    <ReplaceTransition>
      <Transition nodeRef={firstNodeRef} />
      <Transition nodeRef={secondNodeRef} />
    </ReplaceTransition>
  )
}
```

If is using this way, then we can't get that ref

```jsx
function App() {
  return (
    <ReplaceTransition>
      <Fade />
      <Fade />
    </ReplaceTransition>
  )
}

function Fade() {
  const nodeRef = useRef()
  return <Transition nodeRef={nodeRef} />
}
```

Reply:
For posterity: this was discussed over DM, the decision was for `ReplaceTransition` to search for `nodeRef` of direct children, otherwise it falls back to `findDOMNode`.